### PR TITLE
[IMP] spreadsheet: Index function

### PR DESCRIPTION
**Pull request by mamu-odoo**
## Description:

Implementation of  the [Index(reference, [row], [column])](https://support.google.com/docs/answer/3098242?hl=en).
Odoo task ID : [2871406](https://www.odoo.com/web#id=2871406&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_lt("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/functions/module_lookup.ts [hunk 1]
```diff
@@ -213,6 +213,43 @@ export const HLOOKUP: AddFunctionDescription = {
   isExported: true,
 };
 
+// -----------------------------------------------------------------------------
+// INDEX
+// -----------------------------------------------------------------------------
+export const INDEX: AddFunctionDescription = {
+  description: _lt("Returns the content of a cell, specified by row and column offset."),
+  args: [
+    arg("reference (range)", _lt("The range of cells from which the values are returned.")),
```
Comment (laa-odoo) [inline src/functions/module_lookup.ts:222]: Could be any too

File: src/functions/module_lookup.ts [hunk 2]
```diff
@@ -213,6 +213,43 @@ export const HLOOKUP: AddFunctionDescription = {
...
+    arg(
+      "row (number, optional, default=0)",
```
Comment (laa-odoo) [inline src/functions/module_lookup.ts:224]: FYI Optional is not mandatory when default is present

File: src/functions/module_lookup.ts [hunk 3]
```diff
@@ -213,6 +213,43 @@ export const HLOOKUP: AddFunctionDescription = {
...
+      _lt("The index of the row to be returned from within the reference range of cells.")
+    ),
+    arg(
+      "column (number, optional, default=0)",
```
Comment (laa-odoo) [inline src/functions/module_lookup.ts:228]: same here

File: src/functions/module_lookup.ts [hunk 4]
```diff
@@ -213,6 +213,43 @@ export const HLOOKUP: AddFunctionDescription = {
...
+      _lt("The index of the column to be returned from within the reference range of cells.")
+    ),
+  ],
+  returns: ["ANY"],
+  compute: function (reference: MatrixArgValue, row: number = 0, column: number = 0): any {
```
Comment (laa-odoo) [inline src/functions/module_lookup.ts:233]: reference: ArgValue
row: PrimitiveArgValue = 0
column: PrimitiveArgValue = 0

return : FunctionReturnValue

Comment (laa-odoo) [reply src/functions/module_lookup.ts:233]: Need to apply the ```toNumber``` helper on ```row``` and ```column```. And use something like ```_row``` ```_column```  
in the rest of the function

**Review by laa-odoo — CHANGES_REQUESTED** [review]

**Comment by laa-odoo** [comment]
all commits could be squashed :)

**Review by laa-odoo — APPROVED** [review]

File: tests/functions/module_lookup.test.ts [hunk 5]
```diff
@@ -1102,3 +1103,157 @@ describe("XLOOKUP formula", () => {
     });
   });
 });
+
+describe("INDEX formula", () => {
+  const defaultGrid = {
```
Comment (LucasLefevre) [inline tests/functions/module_lookup.test.ts:1108]: since is only used once, it can be in-lined where it's used

**Review by LucasLefevre — COMMENTED** [review]
nice little function :)

**Review by LucasLefevre — APPROVED** [review]
robodoo r+

**Review by LucasLefevre — COMMENTED** [review]
robodoo r+

**Pull request closed without merging**